app/models/collection.rb

```diff
@@ -162,6 +162,54 @@ def has_untranscribed_pages?
     self.works.joins(:work_statistic).where('work_statistics.total_pages > work_statistics.transcribed_pages').exists?
   end
 
+  def update_works_stats
+    works = self.works.includes(:work_statistic)
+    works_stats = get_works_stats_hash
+    works.each do |w|
+      w.work_statistic.recalculate_from_hash(works_stats[w.id]) if works_stats.key?(w.id)
+    end
+    calculate_complete
+  end
+
+  def enable_ocr
+    works.update_all(ocr_correction: true)
+    update_works_stats
+  end
+
+  def disable_ocr
+    works.update_all(ocr_correction: false)
+    update_works_stats
+  end
+
+  def get_works_stats_hash
+    stats = {}
+    work_prototype = {
```

Comment:
If this is immutable, should it be a constant `WORK_PROTOTYPE` or similar?

Reply:
Agreed. Fixed.